eval.go

```diff
@@ -410,6 +410,15 @@ func (d *indexData) List(ctx context.Context, q query.Q) (rl *RepoList, err erro
 		tr.Finish()
 	}()
 
+	if c, ok := q.(*query.RepoRegex); ok {
```

Comment:
I don't think this is the proper place to do this. It feels like it should be in the same places we check `query.Repo`, `query.RepoSet` and `query.RepoBranches`. I believe that is in `indexData.simplify`